gengokit/middlewares/templates/endpoints.go

```diff
@@ -4,21 +4,21 @@ const EndpointsBase = `
 package middlewares
 
 import (
-	"github.com/go-kit/kit/endpoint"
 	svc "{{.ImportPath -}} /generated"
 )
 
-// WrapEndpoint will be called individually for all endpoints defined in
-// the service. Implement this with the middlewares you want applied to
-// every endpoint.
-func WrapEndpoint(in endpoint.Endpoint) endpoint.Endpoint {
-	return in
-}
-
 // WrapEndpoints takes the service's entire collection of endpoints. This
 // function can be used to apply middlewares selectively to some endpoints,
 // but not others, like protecting some endpoints with authentication.
 func WrapEndpoints(in svc.Endpoints) svc.Endpoints {
+
+	// Apply a middleware selectively
+	// in.Echo = endpoint.Middleware(in.Echo)
```

Comment:
An `endpoint.Middleware` is a function that takes an `endpoint` and returns an `endpoint`. So I was trying to demonstrate that you pass an `endpoint` to a function of type `endpoint.Middleware` i.e. `endpoint.Middleware(in.Echo)` and you well get an endpoint, which you set `in.Echo` to.